lib/util/http/host-rules.ts

```diff
@@ -120,10 +121,14 @@ export function applyHostRules(url: string, inOptions: GotOptions): GotOptions {
   return options;
 }
 
-export function getRequestLimit(url: string): number | null {
-  const hostRule = hostRules.find({
-    url,
-  });
-  const limit = hostRule.concurrentRequestLimit;
+export function getConcurrentRequestsLimit(url: string): number | null {
+  const { concurrentRequestLimit: limit } = hostRules.find({ url });
   return typeof limit === 'number' && limit > 0 ? limit : null;
```

Comment:
```suggestion
   return typeof limit === 'number' && limit > 0 ? limit : null;
```
we should use is.number here too,like below to unify code